system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -230,36 +233,25 @@ this.TopStoriesFeed = class TopStoriesFeed {
   }
 
   // If personalization is turned on we have to rotate stories on the client.
-  // An item can only be on top for two iterations (1hr) before it gets moved
-  // to the end. This will later be improved based on interactions/impressions.
+  // Items only be on top for the time configured in settings.
+  // Once an item is expired it will be sorted to the back.
   rotate(items) {
     if (!this.personalized || items.length <= 3) {
       return items;
     }
 
-    if (!this.topItems) {
-      this.topItems = new Map();
-    }
-
-    // This avoids an infinite recursion if for some reason the feed stops
-    // changing. Otherwise, there's a chance we'd be rotating forever to
-    // find an item we haven't displayed on top yet.
-    if (this.topItems.size >= items.length) {
-      this.topItems.clear();
-    }
-
-    const guid = items[0].guid;
-    if (!this.topItems.has(guid)) {
-      this.topItems.set(guid, 0);
-    } else {
-      const val = this.topItems.get(guid) + 1;
-      this.topItems.set(guid, val);
-      if (val >= 2) {
-        items.push(items.shift());
-        this.rotate(items);
+    const maxImpressionAge = (this.recsExpireTime || 5400) * 60 * 1000;
```

Comment:
nit: let's make a constant for `5400 * 60 * 1000`

Reply:
Done, actually a bad bug there too, as 90 mins are 5400 * 1000, not 5400 * 60 * 1000.